include/boost/geometry/algorithms/simplify.hpp

```diff
@@ -71,41 +81,32 @@ struct simplify_range_insert
 
 struct simplify_copy
 {
-    template <typename Range, typename Strategy, typename Distance>
-    static inline void apply(Range const& range, Range& out,
+    template <typename RangeIn, typename RangeOut, typename Strategy, typename Distance>
+    static inline void apply(RangeIn const& range, RangeOut& out,
                              Distance const& , Strategy const& )
     {
         std::copy
             (
-                boost::begin(range), boost::end(range), geometry::range::back_inserter(out)
+                boost::begin(range), boost::end(range),
+                    geometry::range::back_inserter(out)
             );
     }
 };
 
 
-template<std::size_t Minimum>
+template <std::size_t MinimumToUseStrategy>
 struct simplify_range
 {
-    template <typename Range, typename Strategy, typename Distance>
-    static inline void apply(Range const& range, Range& out,
+    template <typename RangeIn, typename RangeOut, typename Strategy, typename Distance>
+    static inline void apply(RangeIn const& range, RangeOut& out,
                     Distance const& max_distance, Strategy const& strategy)
     {
-        // Call do_container for a linestring / ring
-
-        /* For a RING:
-            The first/last point (the closing point of the ring) should maybe
-            be excluded because it lies on a line with second/one but last.
-            Here it is never excluded.
+        // For a RING:
+        // Note that, especially if max_distance is too large,
+        // the output ring might be self intersecting while the input ring is
+        // not, although chances are low in normal polygons
 
-            Note also that, especially if max_distance is too large,
-            the output ring might be self intersecting while the input ring is
-            not, although chances are low in normal polygons
-
-            Finally the inputring might have 3 (open) or 4 (closed) points (=correct),
-                the output < 3 or 4(=wrong)
-        */
-
-        if (boost::size(range) <= int(Minimum) || max_distance < 0.0)
+        if (boost::size(range) <= static_cast<int>(MinimumToUseStrategy) || max_distance < 0)
```

Comment:
`boost::size` returns unsigned type so my guess is that casting is not needed and can cause a comparison warning.